invoices/update.go

```diff
@@ -44,6 +45,14 @@ func (i invoiceUpdateCtx) setID() *[32]byte {
 	return nil
 }
 
+func (i invoiceUpdateCtx) keysendPreimage() *lntypes.Preimage {
+	if i.legacyKeysend {
+		preimage := lntypes.Preimage(i.amp.RootShare())
```

Comment:
Super dope how this all seems to slot in so easily (migration path from keysend to AMP).